File: pdu/python/geometry_msgs/pdu_cdr_conv_Vector3Stamped.py
from .pdu_pytype_Vector3Stamped import Vector3Stamped
from ..pdu_cdr_runtime import CdrReader, CdrWriter

from ..std_msgs.pdu_cdr_conv_Header import *
from ..builtin_interfaces.pdu_cdr_conv_Time import *
from ..geometry_msgs.pdu_cdr_conv_Vector3 import *



def py_to_cdr_body_Vector3Stamped(writer: CdrWriter, src: Vector3Stamped):
    py_to_cdr_body_Header(writer, src.header)
    py_to_cdr_body_Vector3(writer, src.vector)


def cdr_body_to_py_Vector3Stamped(reader: CdrReader, dst: Vector3Stamped):
    cdr_body_to_py_Header(reader, dst.header)
    cdr_body_to_py_Vector3(reader, dst.vector)
    return dst


def py_to_cdr_Vector3Stamped(src: Vector3Stamped) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_Vector3Stamped(writer, src)
    return writer.bytes()


def cdr_to_py_Vector3Stamped(cdr_payload) -> Vector3Stamped:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = Vector3Stamped()
    return cdr_body_to_py_Vector3Stamped(reader, dst)

File: pdu/javascript/geometry_msgs/pdu_cdr_conv_Vector3Stamped.js
import { Vector3Stamped } from './pdu_jstype_Vector3Stamped.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';
import { Header } from '../std_msgs/pdu_jstype_Header.js';
import { PduHeaderConverter } from '../std_msgs/pdu_cdr_conv_Header.js';
import { Time } from '../builtin_interfaces/pdu_jstype_Time.js';
import { PduTimeConverter } from '../builtin_interfaces/pdu_cdr_conv_Time.js';
import { Vector3 } from '../geometry_msgs/pdu_jstype_Vector3.js';
import { PduVector3Converter } from '../geometry_msgs/pdu_cdr_conv_Vector3.js';


export class PduVector3StampedConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { Vector3Stamped } src
     */
    static to_cdr_body(writer, src) {
        PduHeaderConverter.to_cdr_body(writer, src.header);
        PduVector3Converter.to_cdr_body(writer, src.vector);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { Vector3Stamped } dst
     * @returns { Vector3Stamped }
     */
    static cdr_body_to_js(reader, dst) {
        PduHeaderConverter.cdr_body_to_js(reader, dst.header);
        PduVector3Converter.cdr_body_to_js(reader, dst.vector);
        return dst;
    }

    /**
     * @param { Vector3Stamped } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { Vector3Stamped }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new Vector3Stamped());
    }
}
